### ISSUE TITLE
Cluster Tick Process

### ISSUE THREAD

**Issue by bbickerton** [cerana/cerana#258]
#### Issue Description

Create a process that will run periodically to ensure the cluster is in the correct state. This includes creation and destruction of datasets and bundles on nodes. Use the configuration, current heartbeats, and current resource usage from the `clusterconf-provider` to determine what actions need to be taken and where, and then perform the actions. 
- General
  - Check if running on the current consul master. If so, continue. If not, do nothing until the next tick.
  - When deleting a copy of resource, delete the TTL heartbeat of it first
  - Predictable deleting algorithm (e.g. dataset, server that can use the space the most. Perhaps one op at a time to see how usage changes). This is a stretch goal, unnecessary for the demo.
- ZFS
  -  Make sure rw datasets have the correct replication (either by policy or by override for that particular dataset). 
    - Trigger tasks to copy data via zfs stream if not enough replications
    - Trigger tasks to destroy dataset replications If too many copies exist. Do not destroy any copies that are currently in use. Check the bundles running on the node. The replication factor is really a minimum.
- Process Bundles
  - Make sure there are the correct number of replications of a bundle running in the cluster
  - Trigger tasks to run bundles if not enough replications
    - Attempt to place new bundles on machines that have the images preferably already.
    - The daemon should clone the image to new machines otherwise, and add service definition
  - Trigger tasks to destroy bundle replications if too many replications

**Comment by bbickerton** [comment]
Once #349 is merged, what remains:
- Honor replication quantity settings
- Add bundle stopping and dataset delete flagging
